File: areYouSquare/narcissistic.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class narcissistic {
   public static void main(String[] args) {
      System.out.println(isNarcissistic(370));
      System.out.println(isNarcissistic(301));
   }

   public static boolean isNarcissistic(int num) {
      String stringNum = String.valueOf(num);
      String[] strinDigits = stringNum.split("");
      List<Double> powerArr = new ArrayList<Double>();

      int[] numbers = Arrays.stream(strinDigits).mapToInt(Integer::parseInt).toArray(); 
      for(int digit : numbers){
         double power = Math.pow(digit, 3);
         powerArr.add(power);
      }

      double totalPowers = powerArr.stream().mapToDouble(Double::doubleValue).sum();

      return (totalPowers == num) ? true : false;
   }
}

File: areYouSquare/narcissistic.js
isNarcissistic = (num) => {
   num = num.toString()
   let powers = []
   // take in number
   // split the Number
   // foreach digit of number
   // cube the number and store the result
   // add all the cube results
   // if the added result is equal to the original number return true:false

   let digitsArr = num.split("");
   digitsArr.forEach(digit => {
      let power = Math.pow(digit,3)
      powers.push(power);
   });
   powers = powers.reduce((a, b) => a + b, 0);
   return(powers==num)?true:false;
}

console.log(isNarcissistic(301));
console.log(isNarcissistic(370));

// console.log(typeof num)